perceval/backends/jenkins.py

```diff
@@ -0,0 +1,243 @@
+
+    :param site: URL of jenkins node: https://build.opnfv.org/ci
+
+    :raises HTTPError: when an error occurs doing the request
+    """
+
+    def __init__(self, site):
+        self.site = site
+
+
+    def get_jobs(self):
+        """ Retrieve all jobs
+        """
+        url_jenkins = self.site + "/view/All/api/json"
+
+        req = requests.get(url_jenkins)
+        req.raise_for_status()
+        return req.text
+
+    def get_builds(self, job_url):
+        """ Retrieve all builds from a job
+        """
+        # depth=2 to get builds details
+        url_jenkins = job_url + "api/json?depth=2"
+
+        req = requests.get(url_jenkins)
+        req.raise_for_status()
+        return req.text
+
+class JenkinsCommand(BackendCommand):
+    """Class to run Jenkins backend from the command line."""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.site = self.parsed_args.site
+        self.job_url = self.parsed_args.job_url
+        self.origin = self.parsed_args.origin
+        self.outfile = self.parsed_args.outfile
+
+        if not self.parsed_args.no_cache:
+            if not self.parsed_args.cache_path:
+                base_path = os.path.expanduser('~/.perceval/cache/')
+            else:
+                base_path = self.parsed_args.cache_path
+
+            cache_path = os.path.join(base_path, self.site)
+
+            cache = Cache(cache_path)
+
+            if self.parsed_args.clean_cache:
+                cache.clean()
+            else:
+                cache.backup()
+        else:
+            cache = None
+
+        self.backend = Jenkins(self.site, job_url=self.job_url,
+                               cache=cache, origin=self.origin)
+
+    def run(self):
+        """Fetch and print the Builds.
+
+        This method runs the backend to fetch the Builds of a given site.
+        Builds are converted to JSON objects and printed to the
+        defined output.
+        """
+        if self.parsed_args.fetch_cache:
+            builds = self.backend.fetch_from_cache()
+        else:
+            builds = self.backend.fetch()
+
+        try:
+            for build in builds:
+                obj = json.dumps(build, indent=4, sort_keys=True)
+                self.outfile.write(obj)
+                self.outfile.write('\n')
+        except requests.exceptions.HTTPError as e:
+            raise requests.exceptions.HTTPError(str(e.response.json()))
+        except IOError as e:
+            raise RuntimeError(str(e))
+        except Exception as e:
+            if self.backend.cache:
+                self.backend.cache.recover()
+            raise RuntimeError(str(e))
+
+    @classmethod
+    def create_argument_parser(cls):
+        """Returns the Jenkins argument parser."""
+
+        parser = super().create_argument_parser()
+
+        # Jenkins options
+        group = parser.add_argument_group('Jenkins arguments')
+
+        group.add_argument("--site", required=True,
```

Comment:
site now is url, to be consistent with other backends. It is a positional argument.

And "my-job" param is not supported anymore.

Done!